Enable Common Runtime Tests for Mono on Linux-Arm64 in CI

naricc opened the pull request:
Enable runtime tests on Linux Arm-64 with Mono.

steveisok commented on eng/pipelines/runtime.yml:
```diff
@@ -410,6 +410,7 @@ jobs:
     platforms:
     - Android_x86
     - Android_arm64
+    - Linux_arm
```
@akoeplinger do you think we need Linux_arm debug on the libraries build?

akoeplinger replied:
No, the existing build should be enough just like we do for Linux_arm64 today: https://github.com/dotnet/runtime/blob/f007634b02c4484f60ead75a893eb7f89f615ccf/eng/pipelines/runtime.yml#L390

naricc replied:
Ok, switching it to use the release build of libraries.

davidwrighton committed:
parent b70309a5e8cfb89912001ec3d1deeec5f553411b
author David Wrighton <davidwr@microsoft.com> 1588796813 -0700
committer Nathan Ricci <naricc@microsoft.com> 1588797987 -0400

parent b70309a5e8cfb89912001ec3d1deeec5f553411b
author David Wrighton <davidwr@microsoft.com> 1588796813 -0700
committer Nathan Ricci <naricc@microsoft.com> 1588797985 -0400

parent b70309a5e8cfb89912001ec3d1deeec5f553411b
author David Wrighton <davidwr@microsoft.com> 1588796813 -0700
committer Nathan Ricci <naricc@microsoft.com> 1588797980 -0400

parent b70309a5e8cfb89912001ec3d1deeec5f553411b
author David Wrighton <davidwr@microsoft.com> 1588796813 -0700
committer Nathan Ricci <naricc@microsoft.com> 1588797960 -0400

Added ci lane for linux-arm64 on mono.

More updates to issues.targets

Removed whitespace changes, again.

Remvoed trailing whitespace.

More indentation fixes.

Added more archs for runtime tests.

undo unecessary changes.

Changed list of architectures to prioritize most important ones.

Added build for Linux_arm libraries.

got rid of extra arm build.

Added linux_arm, linux_arm64 debug library builds.

remvoed extra arm64 libraries build.

White space.

Removed white space.

Switched to release version of libraries.

Always adding RuntimeFlavor to the libs builds

Changed arm.

Changed where arm is built.

Changed where arm is built.

Moved arm back to release build only.

Added upper case version of liveLibraryBuildConfig.

Commit:
Removed changes that shouldn't be here.

Commit:
Merge branch 'master' of github.com:dotnet/runtime into naricc/more-arcs-runtime-tests

Commit:
Removed extra newline.

Commit:
Added linux_arm64 to test run.

Commit:
Changed to build config upper.

safern commented on eng/pipelines/common/templates/runtimes/run-test-job.yml:
```diff
@@ -274,7 +274,7 @@ jobs:
       - script: $(_msbuildCommand)
                 $(Build.SourcesDirectory)/src/mono/mono.proj
                 /t:PatchCoreClrCoreRoot
-                /p:Configuration=$(buildConfig)
```
I'm surprised this is needed, why are coreclr run jobs working currently?

safern replied:
Oh, this is Mono to patch the runtime, disregard.

safern commented on eng/pipelines/runtime.yml:
```diff
@@ -691,6 +691,7 @@ jobs:
     runtimeFlavor: mono
     platforms:
     - OSX_x64
+    - Linux_arm64
```
Can we open an issue to track removing these steps and using the coreclr build-test steps instead? Of course the best will be to wait for: https://github.com/dotnet/runtime/pull/35783 to land.

naricc replied:
Ok, here it is: https://github.com/dotnet/runtime/issues/35979

safern replied:
Thanks.

naricc commented:
I still need to add exclusions to issues.targets for linux arm64, but the logic is otherwise right and I thought I should start reviews.

Commit:
Added arm64 excludes.

Commit:
Fixed issue with issues.targets

safern commented on src/coreclr/tests/issues.targets:
```diff
@@ -1922,184 +1922,181 @@
-        </ExcludeList>
-        <ExcludeList="**">
+        <ExcludeList Include = "$(XunitTestBinBase)/Regression/VS-ia64-JIT/M00/b108366/b108366/**">
 	    <Issue>needs triage</Issue>
```
`<Issue>` tags are not correctly indented.

Commit:
Fixed indentation.

Commit:
Added alt jit arch.

Commit:
White space.

Commit:
Undid white space.

Commit:
Added missing part of path in issues.targets.

safern commented:
@jashook this will add another test run in Linux_ARM64, so just to be aware and on the lookout for any ARM64 slowdown on our test runs.

Commit:
Got rid of extra x.

Commit:
Excluded another missing test.

Commit:
Added another test to issues.targets.

Commit:
Added missing paren.
